Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -490,6 +490,26 @@ public NodejsProjectNode(NodejsProjectPackage package)
             if (CommonUtils.IsSubpathOf(_intermediateOutputPath, fileNode.Url)) {
                 return false;
             }
+
+            int maxModulesDepth = AnalysisConstants.MaxAnalysisDepthQualty;
+            if (this._analyzer.AnalysisLevel == Options.AnalysisLevel.Medium) {
+                maxModulesDepth = AnalysisConstants.MaxAnalysisDepthFast;
+            }
+
+            var relativeFile = CommonUtils.GetRelativeFilePath(this.FullPathToChildren, fileNode.Url);
+            int nestedModulesCount = 0;
+            int startIndex = 0;
+            int index = relativeFile.IndexOf(AnalysisConstants.NodeModulesFolder, startIndex, StringComparison.OrdinalIgnoreCase);
+            while (index != -1) {
+                nestedModulesCount++;
+                if (nestedModulesCount > maxModulesDepth){
+                    return false;
+                }
+
+                startIndex = index + 1;
+                index = relativeFile.IndexOf(AnalysisConstants.NodeModulesFolder, startIndex, StringComparison.OrdinalIgnoreCase);
+            }
+

```

Comment:
This entire block should be placed below the checks for _analysisIgnoredDirs and _maxFileSize, rather than above because those should take precedent.